pkg/test/extensions/binary.go

```diff
@@ -0,0 +1,380 @@
+package extensions
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	kapierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/openshift/origin/test/extended/util"
+)
+
+type extensionBinaryStruct struct {
+	// The payload image tag in which an external binary path can be found
+	imageTag string
+	// The binary path to extract from the image
+	binaryPath string
+}
+
+var extensionBinaries = []extensionBinaryStruct{
+	{
+		imageTag:   "hyperkube",
+		binaryPath: "/usr/bin/k8s-tests-ext.gz",
+	},
+}
+
+// TestBinary is an abstraction around extracted test binaries that provides an interface for listing the available
+// tests. In the future, it will implement the entire openshift-tests-extension interface.
+type TestBinary struct {
+	path   string
+	logger *log.Logger
+}
+
+// ListTests returns which tests this binary advertises.  Eventually, it should take an environment struct
+// to provide to the binary so it can determine for itself which tests are relevant.
+func (b *TestBinary) ListTests(ctx context.Context) (ExtensionTestSpecs, error) {
```

Comment:
@smg247 Eventually after TRT-1853 would pass the environment stuff into this function